src/main.ts

```diff
@@ -13,5 +13,21 @@ export const MustHaveApiVersion = new SpecificationRule({
 
 export default {
   name: "example-ruleset",
-  rules: [MustHaveApiVersion],
+  description: "An example ruleset that validates things in OpenAPI",
+  // A JSON schema object
+  configSchema: {
+    type: "object",
+    properties: {
+      required_on: {
+        type: "string",
+        enum: ["always", "added"],
+      },
+    },
+  },
+  rulesetConstructor: (options: { required_on: "always" | "added" }) => {
```

Comment:
Ah, ok. So it's just that it's been run through the config, so it could really be whatever custom type they want it to be, but our rulesetConstructor typing would just say it's unknown.

Reply:
I think here if we leave it possible to not validate the schema by not giving us one we could pass the raw payload to their function and if someone wants to validate with their other preferred library they could do so in the constructor implementation. That'd keep us from validating things for them, so there's clearly benefit to doing it our way, but I could see some teams deciding it's better to just handle it on their own.

Reply:
Yeah I think if that's the case the config schema can be empty (or null). I checked that our UI / backend code should work if you just pass in an empty object, we wont' validate but will still allow you to save, which means you could pass in an arbitrary object.

Reply:
I'll update the wording so it's clearer that you can opt out of validation

Reply:
Cool - I'll make sure running against nothing is ok on my side. For now any error during ruleset constructor is just going to add a warning and move on.